### PR TITLE
Make TypeLiteralNode visitor abstract in TexlVisitor

### PR DESCRIPTION
**Pull request by adithyaselv**
Making the `TypeLiteralNode` visitor abstract in the `TexlVisitor` to ensure we consider handling TypeLiteralNode in future TexlVisitor implementations. keeping it virtual sometimes made this go unnoticed.

### CODE REVIEW

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0012 M:Microsoft.PowerFx.Syntax.TexlVisitor.Visit(Microsoft.PowerFx.Syntax.TypeLiteralNode)

File: src/libraries/Microsoft.PowerFx.Core/Syntax/Visitors/TexlVisitor.cs [hunk 1]
```diff
@@ -12,9 +12,7 @@ public abstract class TexlVisitor
         /// Visit <see cref="TypeLiteralNode"/> leaf node.
         /// </summary>
         /// <param name="node">The visited node.</param>
-        public virtual void Visit(TypeLiteralNode node)
-        {
-        }
+        public abstract void Visit(TypeLiteralNode node);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Syntax/Visitors/TexlVisitor.cs:15]: >abstract [](http://example.com/codeflow?start=15&length=8)

adding "Breaking API" tag

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0012 M:Microsoft.PowerFx.Syntax.TexlVisitor.Visit(Microsoft.PowerFx.Syntax.TypeLiteralNode)

**Pull request merged**